pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -99,8 +134,26 @@ func (a *AmbientIndex) insertWorkloadToService(svcAddress string, workload *mode
 	a.byService[svcAddress] = append(a.byService[svcAddress], workload)
 }
 
-func (a *AmbientIndex) updateWaypoint(scope model.WaypointScope, ipStr string, isDelete bool, c *Controller) map[model.ConfigKey]struct{} {
-	addr := netip.MustParseAddr(ipStr).AsSlice()
+func compareGatewayAddress(w1, w2 *workloadapi.GatewayAddress) bool {
+	return gatewayAddressToString(w1) == gatewayAddressToString(w2)
```

Comment:
can't we just use `w1.String()` and `w2.String()` here? `gatewayAddressToString()` seems unnecessary

Reply:
I actually missed that method, it does work - but ultimately I see we have a `proto.Equal()` internal that is used for Workloads already so I just decided to use that